docs/data/date-pickers/custom-components/custom-components.md

```diff
@@ -1,10 +1,10 @@
 ---
 productId: x-date-pickers
-title: Date and Time Pickers - Custom subcomponents
+title: Date and Time Pickers - Custom slots and subcomponents
```

Comment:
What do you think about not renaming the menu item and it's heading? 🤔 
Technically, almost all the `Visual customization` pages are about custom slots. 🙈 

Maybe it's already time to rename all or most of the `components` word mentions on this page into `slots`? 🤔 
Or at least only rename the section below to `Overriding slot components` or `Overidding slots`?

Reply:
I am somewhat in favor of leaving at least some mention of the word 'subcomponent' 🤔 
I am not sure how easy it is for someone who is starting out on this doc to make the connection between `slot` and `subcomponent` 🤔 
I am open to suggestions on this

> Technically, almost all the Visual customization pages are about custom slots

Yeah, that is true. I feel like the other pages are more focused on specific use-cases. This one includes more scenarios and is somewhat of a gateway into understanding slots better. It would still make sense to have the slots mentioned in the menu as well 🤔 

> rename the section below to Overriding slot components or Overidding slots

totally agree 👍

Reply:
Would you agree that eventually, this page will become an home page for slot customization but all the actual examples will be moved to standalone pages (like we started doing)?

Reply:
You are right, it probably will. And I would be in favor of using the term 'components' less and less, but maybe removing 'subcomponents' entirely from the headings is not the way to go 🤔 I feel like it's clear to us that slots are actually building blocks that you can override, but is it clear to users at a first look? So if somebody is on the docs, trying to figure out how to replace the `openPickerIcon`, which menu item is more self-explanatory 'Custom slots' or 'Custom slots and subcomponents'?

Reply:
I agree that "slot" is not understandable by people reaching to the doc for the 1st time

Reply:
What about using a verb to describe what the user would actually do with this information? For example, the Base and Joy docs on this topic use the title "Overriding component structure", which IMO gives the reader a much better idea of how to apply this information, and doesn't require any knowledge of MUI-specific terminology.

Reply:
`Overriding component structure` is definitely a super valid option 💯 But in our case, I feel like it would add a bit of inconsistency, especially since the `Visual customization` section includes a number of other pages, most of them named like this:
![image](https://github.com/mui/mui-x/assets/72460825/8359a1df-02e7-4302-a27d-a3dda221deaa)
I like the idea of introducing the word 'slots' to suggest the link between slots and subcomponents (also helps the search of either of these terms), and revisiting this topic in a broader refactor of the docs structure. WDYT about this?